### DISCUSSION TITLE
Building LLVM 7.0.1 gives warning with fatal error

### DISCUSSION THREAD

**Discussion by hafsajamal987** [intel/llvm#4125] [Q&A]
I am building specfuzz source by following Readme.txt in source https://github.com/tudinfse/SpecFuzz . I am facing error while building specfuzz source.

/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h: In constructor ‘llvm::RegisterScheduler::RegisterScheduler(const char*, const char*, llvm::RegisterScheduler::FunctionPassCtor)’:
/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h:40:54: warning: cast between incompatible function types from ‘llvm::RegisterScheduler::FunctionPassCtor’ {aka ‘llvm::ScheduleDAGSDNodes* (*)(llvm::SelectionDAGISel*, llvm::CodeGenOpt::Level)’} to ‘llvm::MachinePassCtor’ {aka ‘void* (*)()’} [-Wcast-function-type]
   40 |   : MachinePassRegistryNode(N, D, (MachinePassCtor)(C))
      |                                                      ^
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[3]: *** [lib/Target/X86/CMakeFiles/LLVMX86CodeGen.dir/build.make:271: lib/Target/X86/CMakeFiles/LLVMX86CodeGen.dir/X86FrameLowering.cpp.o] Error 1
In file included from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/TargetSubtargetInfo.h:22,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/MachineRegisterInfo.h:32,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/LiveRegUnits.h:19,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/TargetInstrInfo.h:21,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86InstrInfo.h:21,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86FlagsCopyLowering.cpp:26:
/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h: In constructor ‘llvm::RegisterScheduler::RegisterScheduler(const char*, const char*, llvm::RegisterScheduler::FunctionPassCtor)’:
/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h:40:54: warning: cast between incompatible function types from ‘llvm::RegisterScheduler::FunctionPassCtor’ {aka ‘llvm::ScheduleDAGSDNodes* (*)(llvm::SelectionDAGISel*, llvm::CodeGenOpt::Level)’} to ‘llvm::MachinePassCtor’ {aka ‘void* (*)()’} [-Wcast-function-type]
   40 |   : MachinePassRegistryNode(N, D, (MachinePassCtor)(C))
      |                                                      ^
In file included from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/TargetSubtargetInfo.h:22,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/MachineRegisterInfo.h:32,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/LiveRegUnits.h:19,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/TargetInstrInfo.h:21,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86InstrInfo.h:21,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86Subtarget.h:19,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86TargetMachine.h:17,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86TargetMachine.cpp:14:
/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h: In constructor ‘llvm::RegisterScheduler::RegisterScheduler(const char*, const char*, llvm::RegisterScheduler::FunctionPassCtor)’:
/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h:40:54: warning: cast between incompatible function types from ‘llvm::RegisterScheduler::FunctionPassCtor’ {aka ‘llvm::ScheduleDAGSDNodes* (*)(llvm::SelectionDAGISel*, llvm::CodeGenOpt::Level)’} to ‘llvm::MachinePassCtor’ {aka ‘void* (*)()’} [-Wcast-function-type]
   40 |   : MachinePassRegistryNode(N, D, (MachinePassCtor)(C))
      |                                                      ^
In file included from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/TargetSubtargetInfo.h:22,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/MachineRegisterInfo.h:32,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/LiveRegUnits.h:19,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/TargetInstrInfo.h:21,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86InstrInfo.h:21,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86Subtarget.h:19,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86AsmPrinter.h:13,
                 from /home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/src/lib/Target/X86/X86AsmPrinter.cpp:15:
/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h: In constructor ‘llvm::RegisterScheduler::RegisterScheduler(const char*, const char*, llvm::RegisterScheduler::FunctionPassCtor)’:
/home/hafsa/Documents/SpecFuzz-master/llvm-7.0.1/build/include/llvm/CodeGen/SchedulerRegistry.h:40:54: warning: cast between incompatible function types from ‘llvm::RegisterScheduler::FunctionPassCtor’ {aka ‘llvm::ScheduleDAGSDNodes* (*)(llvm::SelectionDAGISel*, llvm::CodeGenOpt::Level)’} to ‘llvm::MachinePassCtor’ {aka ‘void* (*)()’} [-Wcast-function-type]
   40 |   : MachinePassRegistryNode(N, D, (MachinePassCtor)(C))

**Comment by alexbatashev** [comment]
@hafsajamal987  Please, submit DPC++-unrelated issues directly to LLVM community: https://llvm.org/docs/Contributing.html#bug-reports
